src/renderer/kube-object.store.ts

```diff
@@ -224,11 +224,14 @@ export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemSt
 
   @action
   protected updateFromEventsBuffer() {
-    const items = this.items.toJS();
+    const eventsBuffer = this.eventsBuffer.clear();
 
-    for (const { type, object } of this.eventsBuffer.clear()) {
-      const index = items.findIndex(item => item.getId() === object.metadata?.uid);
-      const item = items[index];
+    if (!eventsBuffer.length) {
+      return;
+    }
+
+    for (const { type, object } of eventsBuffer) {
+      const item = this.getById(object.metadata?.uid);
```

Comment:
We don't use the item at all, so we might as well go straight for finding the index.
```suggestion
      const itemIndex = this.getIndexById(object.metadata?.uid);
```